curriculum/challenges/english/01-responsive-web-design/responsive-web-design-principles/create-a-media-query.md

```diff
@@ -33,38 +33,25 @@ Add a media query, so that the `p` tag has a `font-size` of `10px` when the devi
 
 # --hints--
 
-You should declare a `@media` query for devices with a `height` less than or equal to 800px.
+You should declare a `@media` query for devices with a `height` less than or equal to `800px`.
 
 ```js
-assert(
-  $('style')
-    .text()
-    .replace(/\s/g, '')
-    .match(/@media\(max-height:800px\)/g)
-);
+const media = new __helpers.CSSHelp(document).getCSSRules('media');
+assert(media.some(x => x.conditionText?.includes('(max-height: 800px)')));
 ```
 
-Your `p` element should have a `font-size` of 10px when the device `height` is less than or equal to 800px.
+Your `p` element should have a `font-size` of `10px` when the device `height` is less than or equal to `800px`.
 
 ```js
-assert(
-  $('style')
-    .text()
-    .replace(/\s/g, '')
-    .match(/@media\(max-height:800px\){p{font-size:10px;?}}/g)
-);
+const media = new __helpers.CSSHelp(document).getCSSRules('media');
+assert([...media?.find(x => x?.conditionText?.includes('(max-height: 800px)')).cssRules]?.find(x => x.selectorText === 'p')?.style.fontSize === "10px");
 ```
 
-Your `p` element should have an initial `font-size` of 20px when the device `height` is more than 800px.
+Your `p` element should have an initial `font-size` of `20px` when the device `height` is more than `800px`.
 
 ```js
-assert(
-  $('style')
-    .text()
-    .replace(/\s/g, '')
-    .replace(/@media.*}/g, '')
-    .match(/p{font-size:20px;?}/g)
-);
+const ifPFirst = new __helpers.CSSHelp(document).getCSSRules()?.find(x => x?.selectorText === 'p' || x?.media);
+assert(ifPFirst?.style?.fontSize === '20px');
```

Comment:
Note: This will come back as false, if the `x?.media` exists - therefore, `p` with a `fontSize` of `20px` is declared after the media query - what we want